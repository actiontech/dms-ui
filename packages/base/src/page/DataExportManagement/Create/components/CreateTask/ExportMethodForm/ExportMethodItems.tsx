import { useTranslation } from 'react-i18next';
import { ExportMethodEnum } from './index.enum';
import { IconOrderSQLUpload } from 'sqle/src/icon/Order';
import {
  ExportMethodItemStyleWrapper,
  ExportMethodStyleWrapper
} from './style';

const ExportMethodItems: React.FC<{
  value?: ExportMethodEnum;
  onChange?: (value: ExportMethodEnum) => void;
}> = ({ value, onChange }) => {
  const { t } = useTranslation();
  return (
    <ExportMethodStyleWrapper>
      <ExportMethodItemStyleWrapper
        onClick={() => onChange?.(ExportMethodEnum.sql)}
        active={value === ExportMethodEnum.sql}
      >
        <IconOrderSQLUpload />
        <span className="text">
          {t('dmsDataExport.create.form.method.manualInput')}
        </span>
      </ExportMethodItemStyleWrapper>
    </ExportMethodStyleWrapper>
  );
};

export default ExportMethodItems;
