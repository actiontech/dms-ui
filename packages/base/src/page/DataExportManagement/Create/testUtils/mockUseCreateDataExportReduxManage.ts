import { CreateDataExportPageEnum } from '../../../../store/dataExport';
import { ExportMethodEnum } from '../components/CreateTask/ExportMethodForm/index.enum';
import * as useCreateDataExportReduxManage from '../hooks/index.redux';

export const mockCreateDataExportRedux = {
  updateDataExportInfoOpen: false,
  formValues: {
    baseValues: { workflow_subject: 'test', desc: 'desc' },
    sourceValues: { dbService: '121' },
    methodValues: { sql: 'select 1;', exportMethod: ExportMethodEnum.sql }
  },
  pageState: CreateDataExportPageEnum.CREATE_TASK,
  auditLoading: false,
  submitLoading: false,
  taskIDs: ['11111'],
  workflowID: '123',
  updateFormValues: jest.fn(),
  updatePageState: jest.fn(),
  updateAuditLoading: jest.fn(),
  updateSubmitLoading: jest.fn(),
  initModalStatus: jest.fn(),
  updateModalStatus: jest.fn(),
  updateTaskIDs: jest.fn(),
  updateWorkflowID: jest.fn(),
  clearAllState: jest.fn()
};

export const mockUseCreateDataExportReduxManage = (
  data?: Partial<typeof mockCreateDataExportRedux>
) => {
  const spy = jest.spyOn(useCreateDataExportReduxManage, 'default');
  spy.mockImplementation(() => ({ ...mockCreateDataExportRedux, ...data }));
  return spy;
};
