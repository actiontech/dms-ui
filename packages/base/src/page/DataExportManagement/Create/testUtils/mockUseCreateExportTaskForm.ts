import * as useCreateExportTaskForm from '../hooks/useCreateExportTaskForm';

export const mockUseCreateExportTaskFormReturn = {
  baseForm: null as any,
  sourceForm: null as any,
  methodForm: null as any,
  auditLoading: false,
  formatSQLAction: jest.fn(),
  auditAction: jest.fn(),
  resetAllForms: jest.fn()
};

export const mockUseCreateExportTaskForm = (
  data?: Partial<typeof mockUseCreateExportTaskFormReturn>
) => {
  const spy = jest.spyOn(useCreateExportTaskForm, 'default');
  spy.mockImplementation(() => ({
    ...mockUseCreateExportTaskFormReturn,
    ...data
  }));
  return spy;
};
