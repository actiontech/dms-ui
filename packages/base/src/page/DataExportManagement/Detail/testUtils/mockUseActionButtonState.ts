import * as useActionButtonState from '../components/PageHeaderAction/useActionButtonState';

export const mockActionButtonStateData = {
  closeWorkflowButtonMeta: {
    action: jest.fn(),
    loading: false,
    hidden: false
  },
  approveWorkflowButtonMeta: {
    action: jest.fn(),
    loading: false,
    hidden: false
  },
  rejectWorkflowButtonMeta: {
    action: jest.fn(),
    loading: false,
    hidden: false
  },
  executeExportButtonMeta: {
    action: jest.fn(),
    loading: false,
    hidden: false
  }
};

export const mockUseActionButtonState = (
  data?: Partial<typeof mockActionButtonStateData>
) => {
  const spy = jest.spyOn(useActionButtonState, 'default');
  spy.mockImplementation(() => ({ ...mockActionButtonStateData, ...data }));
  return spy;
};
