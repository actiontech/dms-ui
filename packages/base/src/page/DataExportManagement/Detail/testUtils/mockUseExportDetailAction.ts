import * as useExportDetailAction from '../hooks/useExportDetailAction';

export const mockExportDetailActionData = {
  refreshWorkflow: jest.fn(),
  closeWorkflowLoading: false,
  closeWorkflow: jest.fn(),
  approveWorkflowLoading: false,
  approveWorkflow: jest.fn(),
  executeExportLoading: false,
  executeExport: jest.fn()
};

export const mockUseExportDetailAction = (
  data?: Partial<typeof mockExportDetailActionData>
) => {
  const spy = jest.spyOn(useExportDetailAction, 'default');
  spy.mockImplementation(() => ({ ...mockExportDetailActionData, ...data }));
  return spy;
};
