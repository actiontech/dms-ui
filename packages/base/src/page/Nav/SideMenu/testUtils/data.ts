export const mockRecentlyOpenedProjectsData = {
  recentlyProjects: [
    {
      project_id: '1',
      project_name: 'test_project'
    }
  ],
  updateRecentlyProject: jest.fn(),
  currentProjectID: '0'
};
