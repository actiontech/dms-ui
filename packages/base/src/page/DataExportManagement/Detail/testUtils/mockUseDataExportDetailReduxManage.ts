import { GetDataExportTaskStatusEnum } from '@actiontech/shared/lib/api/base/service/common.enum';
import * as useDataExportDetailReduxManage from '../hooks/index.redux';
import { GetDataExportWorkflowResponseData } from '../../../../testUtils/mockApi/dataExport/data';

export const mockDataExportDetailRedux = {
  workflowStepOpen: false,
  workflowRejectOpen: false,
  workflowInfo: GetDataExportWorkflowResponseData,
  taskInfos: [
    {
      task_uid: '1752172791873933312',
      db_info: {
        uid: '1752170583749038080',
        name: 'mysql-1',
        db_type: '',
        database_name: 'test'
      },
      status: GetDataExportTaskStatusEnum.finish,
      export_start_time: '2024-01-30T11:32:15.085+08:00',
      export_end_time: '2024-01-30T11:32:15.085+08:00',
      file_name: 'mysql-1_20240130113114-1752172791873933312.zip',
      audit_result: {
        audit_level: '',
        score: 100,
        pass_rate: 1
      },
      export_type: 'SQL',
      export_file_type: 'CSV'
    },
    {
      task_uid: '1752172791873933313',
      db_info: {
        uid: '1752170583749038080',
        name: 'mysql-1',
        db_type: '',
        database_name: 'test'
      },
      status: GetDataExportTaskStatusEnum.exporting,
      export_start_time: '2024-01-30T11:32:15.085+08:00',
      export_end_time: '2024-01-30T11:32:15.085+08:00',
      file_name: 'mysql-1_20240130113114-1752172791873933313.zip',
      audit_result: {
        audit_level: '',
        score: 100,
        pass_rate: 1
      },
      export_type: 'SQL',
      export_file_type: 'CSV'
    },
    {
      task_uid: '1752172791873933314',
      db_info: {
        uid: '1752170583749038080',
        name: 'mysql-1',
        db_type: '',
        database_name: 'test'
      },
      status: GetDataExportTaskStatusEnum.failed,
      export_start_time: '2024-01-30T11:32:15.085+08:00',
      export_end_time: '2024-01-30T11:32:15.085+08:00',
      file_name: 'mysql-1_20240130113114-1752172791873933314.zip',
      audit_result: {
        audit_level: '',
        score: 100,
        pass_rate: 1
      },
      export_type: 'SQL',
      export_file_type: 'CSV'
    }
  ],
  curTaskID: '1752172791873933312',
  taskStatusNumber: {
    success: 1,
    failed: 0,
    exporting: 0
  },
  canRejectWorkflow: false,
  updateWorkflowStepOpen: jest.fn(),
  updateWorkflowRejectOpen: jest.fn(),
  updateWorkflowInfo: jest.fn(),
  updateTaskInfos: jest.fn(),
  updateCurTaskID: jest.fn(),
  updateTaskStatusNumber: jest.fn(),
  updateCanRejectWorkflow: jest.fn(),
  clearAllDetailState: jest.fn()
};

export const mockUseDataExportDetailReduxManage = (
  data?: Partial<typeof mockDataExportDetailRedux>
) => {
  const spy = jest.spyOn(useDataExportDetailReduxManage, 'default');
  spy.mockImplementation(() => ({ ...mockDataExportDetailRedux, ...data }));
  return spy;
};
