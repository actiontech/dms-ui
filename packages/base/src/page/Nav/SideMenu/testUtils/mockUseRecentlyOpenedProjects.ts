import * as useRecentlyOpenedProjects from '../useRecentlyOpenedProjects';
import { mockRecentlyOpenedProjectsData } from './data';

export const mockUseRecentlyOpenedProjects = (
  data?: Partial<typeof mockRecentlyOpenedProjectsData>
) => {
  const spy = jest.spyOn(useRecentlyOpenedProjects, 'default');
  spy.mockImplementation(() => ({
    ...mockRecentlyOpenedProjectsData,
    ...data
  }));
  return spy;
};
