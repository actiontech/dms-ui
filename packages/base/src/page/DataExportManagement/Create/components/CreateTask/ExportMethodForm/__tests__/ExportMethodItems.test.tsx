import { fireEvent, screen } from '@testing-library/react';
import { superRender } from '../../../../../../../testUtils/customRender';
import ExportMethodItems from '../ExportMethodItems';
import { ExportMethodEnum } from '../index.enum';

describe('test ExportMethodItems', () => {
  it('should match snapshot', () => {
    const changeSpy = jest.fn();
    const { container } = superRender(
      <ExportMethodItems value={ExportMethodEnum.sql} onChange={changeSpy} />
    );

    expect(container).toMatchSnapshot();
    fireEvent.click(screen.getByText('输入SQL语句'));
    expect(changeSpy).toHaveBeenCalledTimes(1);
    expect(changeSpy).toHaveBeenCalledWith(ExportMethodEnum.sql);
  });
});
