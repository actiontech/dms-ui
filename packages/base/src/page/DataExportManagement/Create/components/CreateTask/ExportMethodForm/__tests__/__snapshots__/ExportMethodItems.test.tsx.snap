// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ExportMethodItems should match snapshot 1`] = `
<div>
  <div
    class="css-8qroj2"
  >
    <div
      class="update-type-item-wrapper css-16nz79o"
    >
      <span
        class="custom-icon  css-u3jcp4"
      >
        <svg
          fill="currentColor"
          height="18"
          viewBox="0 0 18 18"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 7.5H3V14.25H15V7.5ZM2.25 2.25H15.75C16.1642 2.25 16.5 2.58579 16.5 3V15C16.5 15.4142 16.1642 15.75 15.75 15.75H2.25C1.83579 15.75 1.5 15.4142 1.5 15V3C1.5 2.58579 1.83579 2.25 2.25 2.25ZM3.75 4.5V6H5.25V4.5H3.75ZM6.75 4.5V6H8.25V4.5H6.75ZM3.75 8.25H6V12H3.75V8.25Z"
            fill="currentColor"
          />
        </svg>
      </span>
      <span
        class="text"
      >
        输入SQL语句
      </span>
      <div
        class="active-icon-wrapper"
      >
        <span
          class="icon-order-upload-checked active-icon css-u3jcp4"
        >
          <svg
            fill="none"
            height="14"
            viewBox="0 0 14 14"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="Frame"
            >
              <path
                d="M3.88879 7.00002L6.22213 9.33335L10.8888 4.66669"
                id="Vector"
                stroke="#FCFBF9"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
            </g>
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;
